components/trace/appmap/classmap/estree/visit.test.mjs

```diff
@@ -70,13 +75,13 @@ assertDeepEqual(
           line: 1,
           column: 19,
           static: true,
-          comments: [" @label-1 foo @label-2 ", "@label-3", "bar"],
+          comments: comments,
           range: [19, 26],
           parameters: [[20, 21]],
-          labels: ["@label-1", "@label-2", "@label-3"],
+          labels: ["label-1", "label-2", "label-3"],
           children: [],
         },
       ],
     },
-  ],
-);
+  ]);
+}
```

Comment:
Will you add one more test, please: a function that has a non-empty comment, but that has no labels in it? Thanks! (If that test already exists, and I just missed, let me know.)

Reply:
The agent works by collecting an array of leading comments. So the comment at line 63 is per requested. If you feel that is not enough I can add a test case of course. However I tend to rely a lot on coverage and not add test cases which do not further improve coverage. This is to reduce change impact.

```js
// leading comment 1
/* leading comment2 */
function f () {}
```

Reply:
So, you're saying that each element in the comments array gets tested as a separate comment, right? If that's the case, I'm satisfied with the tests as-is.

Reply:
Indeed